### PR TITLE
Disable add profile src execution by default to make the build log clean

### PR DESCRIPTION
**Pull request by res-life**
Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
This is to remove lots of the following prints
```
[INFO] --- build-helper-maven-plugin:3.2.0:add-source (add-profile-src-30+) @ rapids-4-spark-api-validation ---
[INFO] Source directory: /home/chong/code/spark-rapids/api_validation/src/main/301+-nondb/scala added.
[INFO] Source directory: /home/chong/code/spark-rapids/api_validation/src/main/301until310-all/scala added.
[INFO] Source directory: /home/chong/code/spark-rapids/api_validation/src/main/301until310-nondb/scala added.
[INFO] Source directory: /home/chong/code/spark-rapids/api_validation/src/main/301until320-all/scala added.
[INFO] Source directory: /home/chong/code/spark-rapids/api_validation/src/main/301until320-nondb/scala added.
[INFO] Source directory: /home/chong/code/spark-rapids/api_validation/src/main/301until330-all/scala added.
[INFO] Source directory: /home/chong/code/spark-rapids/api_validation/src/main/pre320-treenode/scala added.

```

**Comment by res-life** [comment]
build

File: pom.xml [hunk 1]
```diff
@@ -663,7 +663,7 @@
                             <execution>
                                 <id>add-profile-src-31+</id>
                                 <goals><goal>add-source</goal></goals>
-                                <phase>generate-sources</phase>
+                                <phase>none</phase>  <!--Disable it for all submodules, only sql-plugin needs to enable it -->
```
Comment (firestarman) [inline pom.xml:666]: How about removing this whole execution if it was not needed ?

Comment (res-life) [reply pom.xml:666]: You are right. But not sure if submodules will use this in future.  
Let's keep it in the parent pom.

**Review by gerashegalov — APPROVED** [review]
LGTM

In a future PR, it would also be great to get rid of duplicate definitions of executions like  `add-profile-src-30+` identical across profiles

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Review by revans2 — APPROVED** [review]

**Pull request merged**